### ISSUE TITLE
$unwind - cannot use optional includeArrayIndex and/or preserveNullAndEmptyArrays

### ISSUE THREAD

**Issue by amfall** [Studio3T/robomongo#1353]
Hello, I'm using Robomongo version 1.0.0 and I want to use $unwind with the options includeArrayIndex and/or preserveNullAndEmptyArrays. 

my collection:
**db.getCollection('uimsi_d').find().pretty();**
/* 1 */
{
    "_id" : NumberLong(11879110741594227),
    "pi" : 1780,
    "ts" : ISODate("2017-04-01T00:00:00.000Z"),
    "rt" : 0,
    "im" : NumberLong(412012601839524),
    "at" : "AFGAW",
    "ms" : NumberLong(93705621573),
    "c01" : [ 
        0, 
        3, 
        0, 
        0
    ]
}

using $unwind on array "c01" **without** optional includeArrayIndex and/or preserveNullAndEmptyArrays works perfectly:
**db.getCollection('uimsi_d').aggregate( { $unwind:"$c01"}).pretty();**
/* 1 */
{
    "_id" : NumberLong(11879110741594227),
    "pi" : 1780,
    "ts" : ISODate("2017-04-01T00:00:00.000Z"),
    "rt" : 0,
    "im" : NumberLong(412012601839524),
    "at" : "AFGAW",
    "ms" : NumberLong(93705621573),
    "c01" : 0
}

/* 2 */
{
    "_id" : NumberLong(83936704779520558),
    "pi" : 1780,
    "ts" : ISODate("2017-04-01T00:00:00.000Z"),
    "rt" : 0,
    "im" : NumberLong(412012202443653),
    "at" : "AFGAW",
    "c01" : 0
}

/* 3 */
{
    "_id" : NumberLong(155994298817446919),
    "pi" : 1780,
    "ts" : ISODate("2017-04-01T00:00:00.000Z"),
    "rt" : 0,
    "im" : NumberLong(412012202434625),
    "at" : "AFGAW",
    "ms" : NumberLong(93702626256),
    "c01" : 0
}

/* 4 */
{
    "_id" : NumberLong(228051892855375569),
    "pi" : 1780,
    "ts" : ISODate("2017-04-01T00:00:00.000Z"),
    "rt" : 0,
    "im" : NumberLong(412012004095781),
    "at" : "AFGAW",
    "c01" : 0
}

Now trying $unwind on array "c01" **without** optional includeArrayIndex doesn't work:
**db.getCollection('uimsi_d').aggregate( { $unwind:{path:"$c01",includeArrayIndex: "offset"}}).pretty();**

I get following error:
![image](https://cloud.githubusercontent.com/assets/28756202/26154150/a72755a4-3b0e-11e7-9ff2-b9bf3c1b6532.png)

From mongodb documentation 3.2, it shall be supported:
![image](https://cloud.githubusercontent.com/assets/28756202/26154281/27023c94-3b0f-11e7-855c-6e1d4a315ca5.png)

Can you help me figure out what is the issue?
- Is it simply not support by Robomongo 1.0.0?
- Am i using wrong syntax?

Thanks in advance for your time!
Amadou.

**Comment by simsekgokhan** [comment]
Hi @amfall , thanks for reporting the problem. 
I tried to create your case as much as identical, and interestingly I run the same command that is not working for you without problems. 

Here is my test:

This is my simplified document in 'uimsi_d' that I tested with:
```
{
    "item" : "ABC1",
    "c01" : [ 
        "S", 
        "M", 
        "L"
    ]
}
```

This is the outout whe I run the same command as you:
**db.getCollection('uimsi_d').aggregate( { $unwind:{path:"$c01",includeArrayIndex: "offset"}}).pretty();**"
```
/* 1 */
{
    "_id" : ObjectId("591d557a9de8d7d1842694a0"),
    "item" : "ABC1",
    "c01" : "S",
    "offset" : NumberLong(0)
}

/* 2 */
{
    "_id" : ObjectId("591d557a9de8d7d1842694a0"),
    "item" : "ABC1",
    "c01" : "M",
    "offset" : NumberLong(1)
}

/* 3 */
{
    "_id" : ObjectId("591d557a9de8d7d1842694a0"),
    "item" : "ABC1",
    "c01" : "L",
    "offset" : NumberLong(2)
}
```

So, to me your command seems to be ok, maybe there is some weird problems with space or characters. Can you try performing my test?

**Comment by amfall** [comment]
Hi @simsekgokhan 
Thanks for your quick answer.
I did your test and it failed with same error.

Here are the details:
db.createCollection("test", { capped : true, size : 1048576, max : 100 } );

db.test.insertOne( {
    "item" : "ABC1",
    "c01" : [ 
        "S", 
        "M", 
        "L"
    ]
}
);

db.test.find();
/* 1 */
{
    "_id" : ObjectId("591dcff8d13c2ae0e97fbcea"),
    "item" : "ABC1",
    "c01" : [ 
        "S", 
        "M", 
        "L"
    ]
}

db.test.aggregate( { $unwind:{path:"$c01",includeArrayIndex:"offset"} }).pretty();
![image](https://cloud.githubusercontent.com/assets/28756202/26214255/03b350c8-3bfc-11e7-9683-ee7db0fd7c33.png)

I performed the test using Robomongo 1.0.0 and 1.1.0_beta and both failed with same error.
What else can I check?

**Comment by amfall** [comment]
Additional info, I'm using Robomongo 1.0.0 and 1.1.0 on Windows 7:
![image](https://cloud.githubusercontent.com/assets/28756202/26214522/f45b111e-3bfc-11e7-9005-f0e6a8383c87.png)

Can it be the issue?

**Comment by amfall** [comment]
Maybe to mention that following command works:
db.test.aggregate( { $unwind:"$c01" }).pretty();
/* 1 */
{
    "_id" : ObjectId("591dcff8d13c2ae0e97fbcea"),
    "item" : "ABC1",
    "c01" : "S"
}

/* 2 */
{
    "_id" : ObjectId("591dcff8d13c2ae0e97fbcea"),
    "item" : "ABC1",
    "c01" : "M"
}

/* 3 */
{
    "_id" : ObjectId("591dcff8d13c2ae0e97fbcea"),
    "item" : "ABC1",
    "c01" : "L"
}

**Comment by amfall** [comment]
Hi @simsekgokhan 
I think I find out what the problem is. My mongo server is version 3.0.2 and therefore doesn't support features that are available in 3.2 or 3.4.
![image](https://cloud.githubusercontent.com/assets/28756202/26241287/6b6fde7e-3c84-11e7-8e0d-e4b21d535860.png)

So this issue is closed from my side. thanks for the time your prompt support!

**Comment by simsekgokhan** [comment]
@amfall Not a problem, glad to see you found the problem.
